docker-compose.yml

```diff
@@ -20,38 +20,75 @@ services:
     restart: always
     networks:
       - merginmaps
-  server:
-    image: lutraconsulting/merginmaps-backend:2024.2.2
+  server-gunicorn:
+    image: server-gunicorn
+    build:
+      context: ./server
+      dockerfile: Dockerfile
+      args:
+        - "UID=${HOST_UID:-901}"
+        - "GID=${HOST_GID:-901}"
     container_name: merginmaps-server
     restart: always
-    user: 901:999
     volumes:
-      - ./projects:/data
+      - ./projects:/home/mergin/app/data
     env_file:
       - .prod.env
     depends_on:
       - db
       - redis
+    command: ["gunicorn --config config.py application:application"]
+    networks:
+      - merginmaps
+  celery-beat:
+    image: celery-beat
+    build:
+      context: ./server
+      dockerfile: Dockerfile
+      args:
+        - "UID=${HOST_UID:-901}"
+        - "GID=${HOST_GID:-901}"
+    container_name: celery-beat
+    env_file:
+      - .prod.env
+    depends_on:
+      - redis
+      - server-gunicorn
+    command: ["celery -A application.celery beat --loglevel=info"]
+    networks:
+      - merginmaps
+  celery-worker:
+    image: celery-worker
+    build:
+      context: ./server
+      dockerfile: Dockerfile
+      args:
+        - "UID=${HOST_UID:-901}"
+        - "GID=${HOST_GID:-901}"
+    container_name: celery-worker
+    env_file:
+      - .prod.env
+    depends_on:
+      - redis
+      - server-gunicorn
+      - celery-beat
+    command: ["celery -A application.celery worker --loglevel=info"]
     networks:
       - merginmaps
   web:
-    image: lutraconsulting/merginmaps-frontend:2024.2.2
+    image: merginmaps-frontend
+    build:
+      context: ./web-app
+      dockerfile: Dockerfile
     container_name: merginmaps-web
     restart: always
     depends_on:
-      - server
+      - server-gunicorn
     links:
       - db
-    networks:
-      - merginmaps
-  proxy:
```

Comment:
I think it depends on deployment type, I know there are some users already using host nginx as both, proxy and for TLS termination. Also there are users using dockerized proxy as stated in current docker-compose example, so I am bit worried about such changes for inexperienced users.

I agree it could be merged and you guys can certainly do that for your k8s deployments but I would prefer here in public repo to keep is as is even if not optimal. We can instead maybe provide some optimization hints instead?

Reply:
I agree with @varmar05  ... It should be useful to have volume for web-app (a.k.a dashboard) nginx config and upgrade It in custom deployment.